src/processors/transform.rs

```diff
@@ -0,0 +1,293 @@
+#![feature(test)]
+
+extern crate cgmath;
+//extern crate test;
+
+use self::cgmath::{Quaternion, Vector3, Matrix3, Matrix4};
+
+use ecs::{Join, Component, NullStorage, VecStorage, Entity, RunArg, Processor};
+use context::Context;
+use std::sync::{Mutex, Arc};
+use std::collections::{HashMap, VecDeque};
+
+/// Local position and rotation from parent.
+#[derive(Debug, Clone, Copy)]
+pub struct LocalTransform {
+    pub pos: [f32; 3], // translation vector
+    pub rot: [f32; 4], // rotation (radians) vector
+    pub scale: [f32; 3], // scale vector
+    pub parent: Option<Entity>,
+    dirty: bool,
+}
+
+impl LocalTransform {
+    #[inline]
+    pub fn pos(&self) -> [f32; 3] { self.pos }
+    #[inline]
+    pub fn rot(&self) -> [f32; 4] { self.rot }
+    #[inline]
+    pub fn scale(&self) -> [f32; 3] { self.scale }
+    #[inline]
+    pub fn parent(&self) -> Option<Entity> { self.parent }
+    #[inline]
+    pub fn set_pos(&mut self, pos: [f32; 3]) {
+        self.pos = pos;
+        self.dirty = true;
+    }
+    #[inline]
+    pub fn set_rot(&mut self, rot: [f32; 4]) {
+        self.rot = rot;
+        self.dirty = true;
+    }
+    #[inline]
+    pub fn set_scale(&mut self, scale: [f32; 3]) {
+        self.scale = scale;
+        self.dirty = true;
+    }
+    #[inline]
+    pub fn set_parent(&mut self, parent: Option<Entity>) {
+        self.parent = parent;
+        self.dirty = true;
+    }
+
+    #[inline]
+    pub fn matrix(&self) -> [[f32; 4]; 4] {
+        let quat: Matrix3<f32> = Quaternion::from(self.rot).into();
+        let mut matrix: Matrix4<f32> = (&quat * Matrix3::new(self.scale[0], 0.0, 0.0,
+                                                             0.0, self.scale[1], 0.0,
+                                                             0.0, 0.0, self.scale[2])).into();
+        matrix.w = Vector3::from(self.pos).extend(1.0f32);
+        matrix.into()
+    }
+}
+
+impl Default for LocalTransform {
+    fn default() -> Self {
+        LocalTransform {
+            pos: [0.0, 0.0, 0.0],
+            rot: [0.0, 0.0, 0.0, 1.0],
+            scale: [1.0, 1.0, 1.0],
+            parent: None,
+            dirty: true,

```

Comment:
I'm just not sure how it would work correctly. Should we have a vec of `LocalTransform`s/`Transform`s of the previous frame and if they are different then add a dirty component or something?